salt/etcd-discovery/init.sls

```diff
@@ -9,7 +9,7 @@ etcd-discovery-setup:
     - name: curl
   cmd.run:
     - name: curl -L -X PUT {{ etcd_size_uri }} -d value={{ salt.k8s_etcd.get_cluster_size() }}
-    - unless: curl {{ etcd_size_uri }} | grep '"value":"{{ salt.k8s_etcd.get_cluster_size() }}"'
+    - onlyif: curl {{ etcd_size_uri }} | grep '"message":"Key not found"'
```

Comment:
We could also use `"errorCode":100` here... but it looks a bit harder to remember on our `sls` file, but safer...